tests/test_client.py

```diff
@@ -0,0 +1,197 @@
+import unittest
+import boto3
+import docker
+import botor
+import pytest
+
+
+def create_botwo_client(minio1, minio2):
+    profiles = {
+        "test_upload": {
+            "source_bucket_pattern": "bucket-a",
+            "source_key_pattern": "a/*",
+            "mapped_bucket_name": "test-upload",
+            "mapped_prefix": "main/"
+        },
+        "test_copy": {
+            "source_bucket_pattern": "source",
+            "mapped_bucket_name": "test-copy",
+            "mapped_prefix": "main/"
+        },
+        "test_copy2": {
+            "source_bucket_pattern": "test-copy",
+        },
+        "test_delete_objects": {
+            "source_bucket_pattern": "delete-objects*",
+            "mapped_bucket_name": "test-delete-objects",
+            "mapped_prefix": "test/"
+        },
+        "test_paginator": {
+            "source_bucket_pattern": "pagi",
+            "mapped_bucket_name": "test-paginator",
+            "mapped_prefix": "dev/"
+        },
+        "test_list_objects": {
+            "source_bucket_pattern": "list",
+            "mapped_bucket_name": "test-list-objects",
+            "mapped_prefix": "dev/"
+        }
+
+    }
+    client_mapping = {"test_upload": minio1, "test_copy": minio1, "test_copy2": minio1,
+                      "test_delete_objects": minio2, "test_paginator": minio2, "test_list_objects": minio2,
+                      "default": minio2}
+
+    return botor.client(client_mapping, profiles)
+
+
+class Test(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        docker_client = docker.from_env()
+
+        cls.docker1 = docker_client.containers.run(image="minio/minio:RELEASE.2021-06-07T21-40-51Z",
+                                                   command=["minio", "server", "/data"],
+                                                   environment={"MINIO_ROOT_USER": "minioadmin",
+                                                                "MINIO_ROOT_PASSWORD": "minioadmin",
+                                                                "MINIO_UPDATE": "off"},
+                                                   ports={"9000": None}, detach=True)
+
+        cls.docker2 = docker_client.containers.run(image="minio/minio:RELEASE.2021-06-07T21-40-51Z",
+                                                   command=["minio", "server", "/data"],
+                                                   environment={"MINIO_ROOT_USER": "minioadmin",
+                                                                "MINIO_ROOT_PASSWORD": "minioadmin",
+                                                                "MINIO_UPDATE": "off"},
+                                                   ports={"9000": None}, detach=True)
+
+        cls.docker1.reload()
```

Comment:
Are we guaranteed that we need an immediate `reload`, and that after it we are guaranteed to see the port and to have a working minio?

Reply:
Yes